sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,744 @@
+      case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_GET:
+        printf("A twin GET response was received\n");
+        if (message->payloadlen)
+        {
+          printf("Payload:\n%.*s\n", message->payloadlen, (char*)message->payload);
+        }
+        break;
+      // An update to the desired properties with the properties as a JSON payload.
+      case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_DESIRED_PROPERTIES:
+        printf("A twin desired properties message was received\n");
+        printf("Payload:\n%.*s\n", message->payloadlen, (char*)message->payload);
+        az_span twin_span = az_span_init((uint8_t*)message->payload, (int32_t)message->payloadlen);
+
+        // Get the new temperature
+        double desired_temp;
+        parse_desired_temperature_property(twin_span, &desired_temp);
+        current_device_temp = (int32_t)desired_temp;
+
+        // Set the max/min temps if apply
+        if (current_device_temp > device_max_temp)
+        {
+          device_max_temp = current_device_temp;
+        }
+        if (current_device_temp < device_min_temp)
+        {
+          device_min_temp = current_device_temp;
+        }
+
+        // Increment the avg count, add the new temp to the total, and calculate the new avg
+        device_temperature_avg_count++;
+        device_temperature_avg_total += current_device_temp;
+        device_avg_temp = device_temperature_avg_total / (int32_t)device_temperature_avg_count;
+
+        // Send back the max temp
+        send_reported_temperature_property(device_max_temp);
+
+        break;
+      // A response from a twin reported properties publish message. With a successfull update of
+      // the reported properties, the payload will be empty and the status will be 204.
+      case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES:
+        printf("A twin reported properties response message was received\n");
+        break;
+    }
+    printf("Response status was %d\n", twin_response.status);
+  }
+  if (az_succeeded(az_iot_hub_client_methods_parse_received_topic(
+          &client, az_span_init((uint8_t*)topicName, topicLen), &method_request)))
+  {
+    printf("Direct Method arrived\n");
+    if (az_span_is_content_equal(report_method_name_span, method_request.name))
+    {
+      az_span method_response_span = AZ_SPAN_FROM_BUFFER(methods_response_payload);
+      az_span method_payload_span
+          = az_span_init((uint8_t*)message->payload, (int32_t)message->payloadlen);
+
+      // Invoke Method
+      az_result response
+          = report_method(method_payload_span, method_response_span, &method_response_span);
+      (void)response;
+
+      // Send method response with report as JSON payload
+      int rc;
+      if ((rc = send_method_response(&method_request, 200, method_response_span)) != 0)
+      {
+        printf("Unable to send %d response, status %d\n", 200, rc);
+      }
+    }
+    else
+    {
+      // Unsupported Method
```

Comment:
Nit: PnP calls "Methods" as "Commands"
Also please printf() the fact that the unknown method arrived and its name for diagnostics.